## 5.2 실수계산

In [39]:
3.5 + 2.1

5.6

In [40]:
4.3 - 2.7

1.5999999999999996

> 1.6이 나와야 하는데 1.5999999999999996이 나왔음. 컴퓨터는 실수를 표현할 때 오차가 발생하기 때문.  
> 자세한 내용은 [부록 ‎47.7 실수 값의 오차](https://dojang.io/mod/page/view.php?id=2466)를 참조.

In [41]:
1.5 * 3.1

4.65

In [42]:
5.5 / 3.1

1.7741935483870968

In [43]:
4.2 + 5 # 실수와 정수를 함께 계산하면 표현 범위가 넓은 실수로 계산됩니다(실수가 정수보다 표현 범위가 넓습니다).

9.2

In [44]:
float(5) # float(숫자) : 숫자 또는 계산 결과를 강제로 실수로 float

5.0

In [45]:
float(1 + 2) # float(계산식)

3.0

In [46]:
float('5.3') # float('문자열') : 실수 또는 정수로 된 문자열이어야 함.

5.3

> ### float
> 부동소수점(floating point)에서 따왔으며 값을 실수로 만들어줌.  
> 즉, 실수는 float 자료형이며 type에 실수를 넣어보면 <class 'float'>가 나옴.
> type(3.5)
> <class 'float'>


> ### 참고 | 복소수
> 실수부와 허수부로 이루어진 복소수(complex number)도 사용할 수 있다.
> 허수부는 숫자 뒤에 j를 붙입니다(수학에서는 허수를 i로 표현하지만 공학에서는 j를 사용)


In [47]:
1.2+1.3j

(1.2+1.3j)

In [48]:
complex(1.2, 1.3) # 복소수로 만들 때는 complex를 사용

(1.2+1.3j)